File: Components/WinForms/CS/PrintingTutorials/Tutorial1/LessonPreviewControl7.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using DevExpress.XtraPrinting;
using DevExpress.Utils;

namespace XtraPrintingDemos.Tutorial1 {
    public class LessonPreviewControl7 : LessonPreviewControl6 {
        static Dictionary<TutorialHelper.ProductsHelper.ProductColumn, string> columnsNames; 

        static LessonPreviewControl7() {
            columnsNames = new Dictionary<TutorialHelper.ProductsHelper.ProductColumn, string>();
            columnsNames[TutorialHelper.ProductsHelper.ProductColumn.Name] = "Name";
            columnsNames[TutorialHelper.ProductsHelper.ProductColumn.Price] = "Price";
            columnsNames[TutorialHelper.ProductsHelper.ProductColumn.Discontinued] = "Discontinued";
            columnsNames[TutorialHelper.ProductsHelper.ProductColumn.Icon] = "Type";
        }

        public LessonPreviewControl7() {
            link.CreateDetailHeaderArea += CreateDetailHeaderArea;
        }

        static readonly Color headerBackColor = Color.FromArgb(224, 214, 211);

        void CreateDetailHeaderArea(object sender, CreateAreaEventArgs e) {
            e.Graph.Font = new Font("Arial", 10f, FontStyle.Bold);
            e.Graph.BorderWidth = 1f;
            e.Graph.BorderColor = borderColor;
            e.Graph.BackColor = headerBackColor;

            foreach(TutorialHelper.ProductsHelper.ProductColumn column in Enum.GetValues(typeof(TutorialHelper.ProductsHelper.ProductColumn))) {
                string columnName = columnsNames[column];
                RectangleF bounds = TutorialHelper.ProductsHelper.GetBoundsForColumn(column);
                TextBrick brick = e.Graph.DrawString(columnName, bounds);
                brick.HorzAlignment = HorzAlignment.Center;
            }
        }
    }
}

File: Components/WinForms/VB/PrintingTutorials/Tutorial1/LessonPreviewControl7.vb
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports DevExpress.XtraPrinting
Imports DevExpress.Utils

Namespace XtraPrintingDemos.Tutorial1
    Public Class LessonPreviewControl7
        Inherits LessonPreviewControl6

        Private Shared columnsNames As Dictionary(Of TutorialHelper.ProductsHelper.ProductColumn, String)

        Shared Sub New()
            columnsNames = New Dictionary(Of TutorialHelper.ProductsHelper.ProductColumn, String)()
            columnsNames(TutorialHelper.ProductsHelper.ProductColumn.Name) = "Name"
            columnsNames(TutorialHelper.ProductsHelper.ProductColumn.Price) = "Price"
            columnsNames(TutorialHelper.ProductsHelper.ProductColumn.Discontinued) = "Discontinued"
            columnsNames(TutorialHelper.ProductsHelper.ProductColumn.Icon) = "Type"
        End Sub

        Public Sub New()
            AddHandler link.CreateDetailHeaderArea, AddressOf CreateDetailHeaderArea
        End Sub

        Private Shared ReadOnly headerBackColor As Color = Color.FromArgb(224, 214, 211)

        Private Sub CreateDetailHeaderArea(ByVal sender As Object, ByVal e As CreateAreaEventArgs)
            e.Graph.Font = New Font("Arial", 10F, FontStyle.Bold)
            e.Graph.BorderWidth = 1F
            e.Graph.BorderColor = borderColor
            e.Graph.BackColor = headerBackColor

            For Each column As TutorialHelper.ProductsHelper.ProductColumn In System.Enum.GetValues(GetType(TutorialHelper.ProductsHelper.ProductColumn))
                Dim columnName As String = columnsNames(column)
                Dim bounds As RectangleF = TutorialHelper.ProductsHelper.GetBoundsForColumn(column)
                Dim brick As TextBrick = e.Graph.DrawString(columnName, bounds)
                brick.HorzAlignment = HorzAlignment.Center
            Next column
        End Sub
    End Class
End Namespace
